.travis.yml

```diff
@@ -37,46 +37,58 @@ jobs:
       # the bootstrap above is older historically; this way of doing it is newer
       # and also simpler. we should aim to reduce/eliminate the duplication.
       - stage: build
-        name: bootstrap and publishLocal
+        name: build, publishLocal, build again
         if: type = pull_request OR repo != scala/scala
         script:
           - set -e
-          - sbt -warn setupPublishCore generateBuildCharacterPropertiesFile headerCheck publishLocal
+          - sbt setupPublishCore generateBuildCharacterPropertiesFile headerCheck publishLocal
           - STARR=$(sed -n 's/^maven\.version\.number=//p' buildcharacter.properties) && echo $STARR
-          - sbt -Dstarr.version=$STARR -warn setupValidateTest compile
-          # as per discussion on #9332, caching the new STARR only goes halfway.
-          # we should also cache the products of the bootstrapped build. currently all three stages
-          # (build, testAll1, testAll2) end up recompiling core projects like `library` and `compiler`.
-          # fixing that will take some care to satisfy sbt/zinc.
+          - sbt -Dstarr.version=$STARR setupValidateTest compile
         workspaces:
           create:
-            name: published-local
+            name: bootstrapped
             paths:
-              - "$PWD/buildcharacter.properties"
+              # so new STARR will be available
+              - "buildcharacter.properties"
               - "$HOME/.ivy2/local/org.scala-lang"
+              # so build products built using new STARR are kept
+              - "target"
+              - "project/target"
+              - "project/project/target"
+              - "project/project/project/target"
+              - "dist"
+              - "build"
 
-      - stage: testAll
-        name: testAll1
+      - stage: test
+        name: tests (junit, scalacheck, et al)
         if: type = pull_request OR repo != scala/scala
         workspaces:
           use: published-local
```

Comment:
This has to match the name above.  So either keep it as "published-local" or change them all to "bootstrapped".